src/interfaces/plugins.ts

```diff
@@ -679,17 +696,30 @@ module.exports = (theApp: any) => {
     })
 
     if (typeof plugin.registerWithRouter == 'function') {
-      plugin.registerWithRouter(router)
       if (typeof plugin.getOpenApi == 'function') {
         app.setPluginOpenApi(plugin.id, plugin.getOpenApi())
       }
     }
-    app.use(backwardsCompat('/plugins/' + plugin.id), router)
 
     if (typeof plugin.signalKApiRoutes === 'function') {
       app.use('/signalk/v1/api', plugin.signalKApiRoutes(express.Router()))
     }
   }
+
+  function doRegisterWithRouter(app: any, plugin: PluginInfo, router: Router) {
+    router.use((req, res, next) => {
```

Comment:
I think it would be more logical to have the middleware inside the `registerWithRouter` test, so that with no `registerWithRouter` the middleware would not be there.

Reply:
That makes more sense.